eric_sse/queue.py

```
@@ -17,57 +17,31 @@ def pop(self) -> MessageContract:
-        """
-        """
+    async def pop(self) -> MessageContract:
         try:
-            with Lock():
-                return self.__messages.pop(0)
+            lock = Lock()
+            async with lock:
+                m = self.__messages.pop(0)
+                return m
         except IndexError:
             raise NoMessagesException
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)